src/libraries/System.Runtime/tests/System/DateTimeOffsetTests.cs

```diff
@@ -74,6 +74,19 @@ public static void Ctor_DateTime_Invalid()
             AssertExtensions.Throws<ArgumentOutOfRangeException>("microsecond", () => new DateTimeOffset(new DateTime(max.Year, max.Month, max.Day, max.Hour, max.Minute, max.Second, max.Millisecond, max.Microsecond + 1, DateTimeKind.Utc)));
         }
 
+        [Fact]
+        public static void Ctor_DateOnly_TimeOnly_TimeSpan()
+        {
+            var dateTimeOffset = new DateTimeOffset(DateOnly.MinValue, TimeOnly.MinValue, TimeSpan.FromHours(-14));
+            VerifyDateTimeOffset(dateTimeOffset, 1, 1, 1, 0, 0, 0, 0, 0, TimeSpan.FromHours(-14));
+            
+            dateTimeOffset = new DateTimeOffset(DateOnly.MaxValue, TimeOnly.MaxValue, TimeSpan.FromHours(14));
+            VerifyDateTimeOffset(dateTimeOffset, 9999, 12, 31, 23, 59, 59, 999, 999, TimeSpan.FromHours(14), 900);
+
+            dateTimeOffset = new DateTimeOffset(new DateOnly(2012, 12, 31), new TimeOnly(13, 50, 10), TimeSpan.Zero);
+            VerifyDateTimeOffset(dateTimeOffset, 2012, 12, 31, 13, 50, 10, 0, 0, TimeSpan.Zero);
```

Comment:
can we add a case like:

```C#
DateTimeOffset now = DateTimeOffset .Now;
DateTimeOffset constructed = new DateTimeOffset(DateOnly.FromDateTime(now.DateTime), (TimeOnly.FromDateTime(now.DateTime), now.Offset);

Assert.Equal(now, constructed);
```

Reply:
If we use `DateTimeOffset.Now` isn't the test determistic anymore?

Reply:
Why? if the test fail you can print the dateTimeOffset value. What is wrong in that?